src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -126,6 +128,9 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
                         var platformPropertyInstance = ProjectPropertyInstance.Create(PlatformMetadataName, selectedPlatform);
                         referenceGlobalProperties[PlatformMetadataName] = platformPropertyInstance;
                     }
+                    // We will use this property to designate an inner build for platform negotiation 
+                    var platformInnerBuildPropertyInstance = ProjectPropertyInstance.Create("PlatformNegotiationInnerBuild", "true");
```

Comment:
Doesn't this cause another evaluation? 

Say project A is x64;x86 and it builds both, AND project B references A and tells A to build x64. Wouldn't we get two evals of effectively the same instance? One with the inner build global prop/platform, and one with just the platform?

Reply:
this is a good point. in that cause we could identify a innerbuild by the lack of PlatformNegotiationouterBuild. that way we dont need to set anything extra and there is no dual eval

Reply:
pushed a change